horovod/common/operations.cc

```diff
@@ -2013,7 +2138,71 @@ bool RunLoopOnce(HorovodGlobalState& state, bool is_coordinator) {
           }
         }
       }
+        else if (response.response_type() == MPIResponse::ResponseType::ALLGATHER){
+          // Attempt to add more responses to this fused response.
+          auto& entry = state.tensor_table[response.tensor_names()[0]];
+
+          // Copy tensor sizes from the MPI response into a vector of int64_t
+          // and compute total size.  This is size of first dimension.
+          int64_t total_dimension_size = 0;
+          for (auto sz : response.tensor_sizes()) {
+            total_dimension_size += sz;
+          }
 
+          // Every tensor participating in Allgather operation may have different
+          // first dimension size, but the rest of dimensions are same for all
+          // tensors.  Here we get shape of tensor sliced by first dimension.
+          // Allgather output will have shape of:
+          // (sum of first dimension of every tensor) x (tensor slice shape).
+          int64_t total_count_of_output_entries = total_dimension_size;
+          for (int i = 1; i < entry.tensor->shape().dims(); ++i) {
+            total_count_of_output_entries *= entry.tensor->shape().dim_size(i);
+          }
+          int element_size;
+          MPI_Type_size(GetMPIDataType(entry.tensor), &element_size);
+          int64_t total_byte_size_of_output =
+              total_count_of_output_entries * element_size;
+
+          while (!responses.empty()) {
+
+            auto new_response = responses.front();
+            assert(new_response.tensor_names().size() == 1);
+            auto& new_entry = state.tensor_table[new_response.tensor_names()[0]];
+
+            int64_t new_total_dimension_size = 0;
+            for (auto sz : new_response.tensor_sizes()) {
+              new_total_dimension_size += sz;
+            }
+            int64_t new_total_count_of_output_entries = new_total_dimension_size;
+            for (int i = 1; i < new_entry.tensor->shape().dims(); ++i) {
+              new_total_count_of_output_entries *= new_entry.tensor->shape().dim_size(i);
+            }
+
+            int new_element_size;
+            MPI_Type_size(GetMPIDataType(new_entry.tensor), &new_element_size);
+            int64_t new_total_byte_size_of_output =
+                new_total_count_of_output_entries * new_element_size;
+
+              if (response.response_type() == new_response.response_type() &&
```

Comment:
Can you re-format using `clang-format`?  Also, we typically don't add space in the beginning of the loop, etc.